### PR TITLE
Fix: Include catalog in table name if defined in model

### PR DESCRIPTION
**Pull request by eakmanrq**

### CODE REVIEW

File: sqlmesh/core/snapshot/definition.py [hunk 1]
```diff
@@ -875,7 +875,14 @@ def _ensure_categorized(self) -> None:
 
 def table_name(physical_schema: str, name: str, version: str, is_temp: bool = False) -> str:
```
Comment (tobymao) [inline sqlmesh/core/snapshot/definition.py:876]: shouldn't this take a catalog ?

Comment (eakmanrq) [reply sqlmesh/core/snapshot/definition.py:876]: Well the solution is to take the catalog from the name if it is defined.

Comment (eakmanrq) [reply sqlmesh/core/snapshot/definition.py:876]: Are you saying as a way to override it?

Comment (tobymao) [reply sqlmesh/core/snapshot/definition.py:876]: ok nvm i got it now i think

**Review by tobymao — APPROVED** [review]

**Pull request merged**